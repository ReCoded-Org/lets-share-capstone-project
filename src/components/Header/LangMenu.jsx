import * as React from "react";
import { Fragment } from "react";
import Link from "next/link";
import Image from "next/image";

import { Menu, Transition } from "@headlessui/react";
const LangMenu = () => {
    return (
        <Menu as='div' className='relative  ml-3'>
            <div>
                <Menu.Button className=' z-1 mr-4 flex flex-row-reverse rounded-full text-sm md:mr-0 md:flex-row'>
                    <span className='sr-only'>Open user menu</span>
                    <img
                        className='h-8 w-8 rounded-full '
                        src={`/images/enFlag.svg`}
                        alt=''
                    />
                    <Image
                        className='arrow absolute -z-10 rotate-90 md:rotate-0 '
                        height='30px'
                        width='20px'
                        src='/images/arrowIcon.svg'
                    />
                </Menu.Button>
            </div>
            <Transition
                as={Fragment}
                enter='transition ease-out duration-100'
                enterFrom='transform opacity-0 scale-95'
                enterTo='transform opacity-100 scale-100'
                leave='transition ease-in duration-75'
                leaveFrom='transform opacity-100 scale-100'
                leaveTo='transform opacity-0 scale-95'
            >
                <Menu.Items className=' bg-white ring-black absolute right-16 top-1 z-50  mt-1 w-24 rounded-md py-1 shadow-lg md:right-0 md:top-10'>
                    <Menu.Item>
                        <Link href='/' locale='en'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/enFlag.svg'
                                />
                                <a>English</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href='/' locale='ar'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor  hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/arFlag.svg'
                                />
                                <a>عربي</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href='/' locale='tr'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/trFlag.svg'
                                />
                                <a>Türkçe</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href='/' locale='ja'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/jaFlag.svg'
                                />
                                <a>日本語</a>
                            </div>
                        </Link>
                    </Menu.Item>
                </Menu.Items>
            </Transition>
        </Menu>
    );
};
export default LangMenu;
