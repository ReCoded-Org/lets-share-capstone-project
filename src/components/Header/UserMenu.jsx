import * as React from "react";
import { Fragment } from "react";
import { Menu, Transition } from "@headlessui/react";
const UserMenu = () => {
    return (
        <Menu as='div' className='relative ml-3'>
            <div>
                <Menu.Button className='mr-4 mb-2 flex rounded-full text-sm md:mr-1 md:mb-0  '>
                    <span className='sr-only'>Open user menu</span>
                    <img
                        className='h-8 w-8  rounded-full'
                        src='/images/noUserPic.svg'
                        alt=''
                    />
                </Menu.Button>
            </div>

            <Transition
                as={Fragment}
                enter='transition ease-out duration-100'
                enterFrom='transform opacity-0 scale-95'
                enterTo='transform opacity-100 scale-100'
                leave='transition ease-in duration-75'
                leaveFrom='transform opacity-100 scale-100'
                leaveTo='transform opacity-0 scale-95'
            >
                <Menu.Items className='bg-white absolute right-14 top-1 z-50 mt-1 flex w-24 flex-col rounded-md py-1 shadow md:right-0 md:top-10'>
                    <Menu.Item>
                        <a
                            href='#'
                            className='my-1   px-2 text-center text-sm text-fontColor hover:bg-btnBgHover'
                        >
                            Login
                        </a>
                    </Menu.Item>
                    <Menu.Item>
                        <a
                            href='#'
                            className=' my-1  px-2 text-center  text-sm text-fontColor hover:bg-btnBgHover'
                        >
                            Sign up
                        </a>
                    </Menu.Item>
                    <Menu.Item>
                        <a
                            href='#'
                            className='my-1   px-2 text-center  text-sm text-fontColor hover:bg-btnBgHover'
                        >
                            Logout
                        </a>
                    </Menu.Item>
                </Menu.Items>
            </Transition>
        </Menu>
    );
};
export default UserMenu;
