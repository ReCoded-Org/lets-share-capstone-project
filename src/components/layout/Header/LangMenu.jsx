import * as React from "react";
import { Fragment } from "react";
import Link from "next/link";
import Image from "next/image";
import { useRouter } from "next/router";
import { Menu, Transition } from "@headlessui/react";

const LangMenu = () => {
    const { asPath } = useRouter();
    const { locale } = useRouter();
    return (
        <Menu as='div' className='relative  ml-3'>
            <div>
                <Menu.Button className=' z-1 mr-4 flex flex-row-reverse rounded-full text-sm md:mr-0 md:flex-row'>
                    <span className='sr-only'>Open user menu</span>
                    <img
                        className='h-8 w-8 rounded-full '
                        src={`/images/${locale}Flag.svg`}
                        alt=''
                    />
                    <Image
                        className='arrow  rotate-90 md:rotate-0 '
                        height='30px'
                        width='20px'
                        src='/images/arrowIcon.svg'
                    />
                </Menu.Button>
            </div>
            <Transition
                as={Fragment}
                enter='transition ease-out duration-100'
                enterFrom='transform opacity-0 scale-95'
                enterTo='transform opacity-100 scale-100'
                leave='transition ease-in duration-75'
                leaveFrom='transform opacity-100 scale-100'
                leaveTo='transform opacity-0 scale-95'
            >
                <Menu.Items className=' ring-black absolute right-16 top-1 z-50 mt-1  w-24 rounded-md bg-[white] py-1 shadow-lg md:right-0 md:top-10'>
                    <Menu.Item>
                        <Link href={asPath} locale='en'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/enFlag.svg'
                                />
                                <a>English</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href={asPath} locale='ar'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor  hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/arFlag.svg'
                                />
                                <a>عربي</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href={asPath} locale='tr'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/trFlag.svg'
                                />
                                <a>Türkçe</a>
                            </div>
                        </Link>
                    </Menu.Item>
                    <Menu.Item>
                        <Link href={asPath} locale='ja'>
                            <div className='my-1 flex cursor-pointer justify-between  px-2 text-sm text-fontColor hover:bg-btnBgHover'>
                                <Image
                                    className='m-5 rounded-full'
                                    height='1'
                                    width='20px'
                                    src='/images/jaFlag.svg'
                                />
                                <a>日本語</a>
                            </div>
                        </Link>
                    </Menu.Item>
                </Menu.Items>
            </Transition>
        </Menu>
    );
};
export default LangMenu;
