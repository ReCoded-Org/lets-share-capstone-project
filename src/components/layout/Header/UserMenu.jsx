import * as React from "react";
import { Fragment } from "react";
import Image from "next/image";
import { Menu, Transition } from "@headlessui/react";
import { useTranslation } from "next-i18next";
import { useAuth } from "context/AuthContext.js";
import { useRouter } from "next/router";
import { auth } from "firebaseConfig";
import Link from "next/link";
const UserMenu = () => {
    const { t } = useTranslation("common");

    const { user, logout } = useAuth();
    const router = useRouter();

    return (
        <Menu as='div' className='relative ml-2'>
            <div>
                <Menu.Button className='mr-4 flex flex-row-reverse items-center rounded-full text-sm md:mr-1 md:mb-0 md:flex-row  '>
                    <span className='sr-only'>Open user menu</span>
                    <span className='mx-3'>
                        {auth?.currentUser?.displayName
                            ? auth?.currentUser?.displayName
                            : ""}
                    </span>
                    <Image
                        src={
                            auth?.currentUser?.photoURL
                                ? auth?.currentUser?.photoURL
                                : "/profile.png"
                        }
                        width={40}
                        height={40}
                        className=' h-8 w-8  rounded-full'
                        alt=''
                    />
                </Menu.Button>
            </div>

            <Transition
                as={Fragment}
                enter='transition ease-out duration-100'
                enterFrom='transform opacity-0 scale-95'
                enterTo='transform opacity-100 scale-100'
                leave='transition ease-in duration-75'
                leaveFrom='transform opacity-100 scale-100'
                leaveTo='transform opacity-0 scale-95'
            >
                <Menu.Items className='absolute right-20 top-1 z-50 mt-1 flex w-28 flex-col rounded-md border border-fontColor/20 bg-[white] py-1 shadow-lg md:right-0 md:top-10'>
                    <Menu.Item>
                        <span className='border-b-2 border-dotted border-[#aaaaaa]  p-1 text-center text-xs'>
                            {auth?.currentUser?.displayName
                                ? auth?.currentUser?.displayName
                                : "sign in"}
                        </span>
                    </Menu.Item>
                    {user && (
                        <Menu.Item>
                            <Link href='/my-profile'>
                                <a
                                    href=''
                                    className='py-2 px-2 text-center text-sm text-fontColor hover:bg-btnBgHover'
                                >
                                    {t("common.myProfile")}
                                </a>
                            </Link>
                        </Menu.Item>
                    )}
                    {!user && (
                        <Menu.Item>
                            <Link href='/login'>
                                <a
                                    href=''
                                    aria-disabled='true'
                                    className='py-2   px-2 text-center text-sm text-fontColor hover:bg-btnBgHover'
                                >
                                    {t("common.logIn")}
                                </a>
                            </Link>
                        </Menu.Item>
                    )}
                    <hr className='mx-2 border-fontColor/20' />
                    {!user && (
                        <Menu.Item>
                            <Link href='/signup'>
                                <a
                                    href=''
                                    className=' py-2  px-2 text-center  text-sm text-fontColor hover:bg-btnBgHover'
                                >
                                    {t("common.signUp")}
                                </a>
                            </Link>
                        </Menu.Item>
                    )}
                    <hr className='mx-2 border-fontColor/20' />
                    {user && (
                        <Menu.Item>
                            <a
                                onClick={async () => {
                                    router.push("/login").then;
                                    await logout();
                                }}
                                href=''
                                className='py-2   px-2 text-center  text-sm text-fontColor hover:bg-btnBgHover'
                            >
                                {t("common.Logout")}
                            </a>
                        </Menu.Item>
                    )}
                </Menu.Items>
            </Transition>
        </Menu>
    );
};
export default UserMenu;
