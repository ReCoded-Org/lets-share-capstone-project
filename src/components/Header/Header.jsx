import { useTranslation } from "react-i18next";
import Link from "next/link";
import { Disclosure } from "@headlessui/react";
import { MenuIcon, XIcon } from "@heroicons/react/outline";
import UserMenu from "./UserMenu";
import LangMenu from "./LangMenu";
const navigation = [
    { name: "Home", href: "/home", current: false },
    { name: "About us", href: "/aboutus", current: false },
    { name: "Goods", href: "/goods", current: false },
    { name: "Blogs", href: "/blogs", current: false },
];
export function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
}
export default function Header() {
    return (
        <Disclosure as='nav' className='bg-transperent'>
            {({ open }) => (
                <>
                    <div className=' mx-auto mb-2 max-w-7xl px-2 sm:px-3 lg:px-4'>
                        <div className=' flex items-center justify-end'>
                            <div className='absolute flex items-center md:hidden'>
                                {/* Mobile menu button*/}
                                <Disclosure.Button className='text-gray-400 hover:text-slate-900 focus:ring-white inline-flex items-center justify-center rounded-md p-2  hover:bg-btnBgHover focus:outline-none focus:ring-2 focus:ring-inset'>
                                    <span className='sr-only'>
                                        Open main menu
                                    </span>
                                    {open ? (
                                        <XIcon
                                            className='block h-6 w-6'
                                            aria-hidden='true'
                                            stroke='#97dbbf'
                                        />
                                    ) : (
                                        <MenuIcon
                                            className='block h-6 w-6'
                                            aria-hidden='true'
                                            stroke='#97dbbf'
                                        />
                                    )}
                                </Disclosure.Button>
                                <div className='hidden sm:ml-6 md:block'>
                                    <div className='flex space-x-4'>
                                        {navigation.map((item) => (
                                            <Link
                                                key={item.name}
                                                href={item.href}
                                            >
                                                <a
                                                    className={classNames(
                                                        item.current
                                                            ? " text-white text-fontColor"
                                                            : " hover:bg-[gray-700] hover:text-white",
                                                        "rounded-md px-3 py-2 font-medium"
                                                    )}
                                                    aria-current={
                                                        item.current
                                                            ? "page"
                                                            : undefined
                                                    }
                                                >
                                                    {item.name}
                                                </a>
                                            </Link>
                                        ))}
                                    </div>
                                </div>
                            </div>
                            <img
                                className='h-14 w-auto'
                                src='/images/dummyLogo.svg'
                                alt='Lets share logo'
                            />
                            <div className='flex flex-1 items-center justify-center sm:items-stretch sm:justify-end'>
                                <div className='flex flex-shrink-0 items-center'>
                                    <div className='hidden sm:ml-6 md:block'>
                                        <div className='flex space-x-4'>
                                            {navigation.map((item) => (
                                                <Link
                                                    key={item.name}
                                                    href={item.href}
                                                >
                                                    <a
                                                        className='hover:text-white rounded-md px-3
                                                        py-2 text-sm font-medium text-fontColor hover:bg-btnBgHover'
                                                        aria-current={
                                                            item.current
                                                                ? "page"
                                                                : undefined
                                                        }
                                                    >
                                                        {item.name}
                                                    </a>
                                                </Link>
                                            ))}
                                        </div>
                                    </div>
                                    <div className='hidden md:block'>
                                        <UserMenu />
                                    </div>
                                    <div className='hidden md:block'>
                                        <LangMenu />
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <Disclosure.Panel className='md:hidden'>
                        <div className='space-y-1 px-2 pt-2 pb-3'>
                            <div className='flex justify-end'>
                                <UserMenu />
                            </div>
                            <div className='flex justify-end'>
                                <LangMenu />
                            </div>
                            {navigation.map((item) => (
                                <Disclosure.Button
                                    key={item.name}
                                    as='a'
                                    href={item.href}
                                    className='-z-20 block rounded-md px-3 py-2 text-end font-medium text-fontColor hover:bg-btnBgHover'
                                    aria-current={
                                        item.current ? "page" : undefined
                                    }
                                >
                                    {item.name}
                                </Disclosure.Button>
                            ))}
                        </div>
                    </Disclosure.Panel>
                </>
            )}
        </Disclosure>
    );
}
