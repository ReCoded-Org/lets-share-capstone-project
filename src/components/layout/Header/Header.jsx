import { useState, useEffect } from "react";
import Link from "next/link";
import { Disclosure } from "@headlessui/react";
import { MenuIcon, XIcon } from "@heroicons/react/outline";
import UserMenu from "./UserMenu";
import LangMenu from "./LangMenu";
import Image from "next/image";

const navigation = [
    { name: "Home", href: "/", current: false },
    { name: "About us", href: "/aboutus", current: false },
    { name: "Goods", href: "/goods", current: false },
    { name: "Blogs", href: "/blogs", current: false },
];
export function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
}
export default function Header() {
    const [animateHeader, setAnimateHeader] = useState(false);

    useEffect(() => {
        const listener = () =>
            window.scrollY > 140
                ? setAnimateHeader(true)
                : setAnimateHeader(false);
        window.addEventListener("scroll", listener);
        return () => {
            window.removeEventListener("scroll", listener);
        };
    }, []);

    return (
        <div className='relative '>
            <div className=' absolute -z-10 h-96 w-full'>
                {/*<Image 
            src='/Vector.png'
            layout="responsive"
            width={1200}
            height={900}
            className=" absolute  -z-1"
            alt="vector"
            /> */}
            </div>

            <Disclosure as='nav' className='bg-transperent  '>
                {({ open }) => (
                    <>
                        <div
                            className={`max-w-auto  opacity-1 top-0 z-30 mb-2   px-2 transition duration-1000   ease-in-out sm:px-3 lg:px-4 ${
                                animateHeader && "bg-[#ddd] opacity-0"
                            }`}
                        >
                            <div className=' flex items-center justify-end'>
                                <div className='absolute flex  items-center md:hidden'>
                                    {/* Mobile menu button*/}
                                    <Disclosure.Button className=' hover:text-slate-900 focus:ring-white inline-flex items-center justify-center rounded-md p-2  hover:bg-btnBgHover focus:outline-none focus:ring-2 focus:ring-inset'>
                                        <span className='sr-only'>
                                            Open main menu
                                        </span>
                                        {open ? (
                                            <XIcon
                                                className='w-10y block h-10'
                                                aria-hidden='true'
                                                stroke='#97dbbf'
                                            />
                                        ) : (
                                            <MenuIcon
                                                className='block h-10 w-10'
                                                aria-hidden='true'
                                                stroke='#97dbbf'
                                            />
                                        )}
                                    </Disclosure.Button>
                                    <div className='hidden sm:ml-6 md:block '>
                                        <div className='flex space-x-4  '>
                                            {navigation.map((item) => (
                                                <Link
                                                    key={item.name}
                                                    href={item.href}
                                                >
                                                    <a
                                                        className={classNames(
                                                            item.current
                                                                ? "  text-[white]"
                                                                : " hover:bg-[gray-700] hover:text-white text-[white]",
                                                            "rounded-md px-3 py-2 font-medium "
                                                        )}
                                                        aria-current={
                                                            item.current
                                                                ? "page"
                                                                : undefined
                                                        }
                                                    >
                                                        {item.name}
                                                    </a>
                                                </Link>
                                            ))}
                                        </div>
                                    </div>
                                </div>
                                <Image
                                    className='h-14 w-auto'
                                    src='/images/dummyLogo.svg'
                                    alt='Lets share logo'
                                    width={80}
                                    height={80}
                                />
                                <div className='flex flex-1 items-center justify-center sm:items-stretch sm:justify-end '>
                                    <div className='flex flex-shrink-0 items-center'>
                                        <div className='hidden sm:ml-6 md:block '>
                                            <div className='mr-5 flex space-x-4 '>
                                                {navigation.map((item) => (
                                                    <Link
                                                        key={item.name}
                                                        href={item.href}
                                                    >
                                                        <a
                                                            className='hover:text-white mx-1 rounded-md px-3
                                                        py-2 text-lg  font-medium text-fontColor hover:bg-btnBgHover '
                                                            aria-current={
                                                                item.current
                                                                    ? "page"
                                                                    : undefined
                                                            }
                                                        >
                                                            {item.name}
                                                        </a>
                                                    </Link>
                                                ))}
                                            </div>
                                        </div>
                                        <div className='hidden md:block  '>
                                            <UserMenu />
                                        </div>
                                        <div className='hidden md:block'>
                                            <LangMenu />
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <Disclosure.Panel className='bg-[white] md:hidden'>
                            <div className='mx-10  space-y-1  rounded-lg px-2  shadow-lg   '>
                                <div className='flex cursor-pointer justify-center hover:bg-[#dce6ef] '>
                                    <UserMenu />
                                </div>
                                <div className='flex cursor-pointer justify-center border-t border-[rgba(179,178,187,0.4)] hover:bg-[#dce6ef]'>
                                    <LangMenu />
                                </div>
                                {navigation.map((item) => (
                                    <Disclosure.Button
                                        key={item.name}
                                        as='a'
                                        href={item.href}
                                        className='-z-20 mt-0  block border-t border-[rgba(179,178,187,0.4)] px-3 py-2 text-center font-medium text-fontColor hover:bg-btnBgHover'
                                        aria-current={
                                            item.current ? "page" : undefined
                                        }
                                    >
                                        {item.name}
                                    </Disclosure.Button>
                                ))}
                            </div>
                        </Disclosure.Panel>
                    </>
                )}
            </Disclosure>
        </div>
    );
}
