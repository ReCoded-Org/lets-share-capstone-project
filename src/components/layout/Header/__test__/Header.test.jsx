"use strict";

import renderer from "react-test-renderer";
import Header from "../Header";
import * as nextRouter from "next/router";
it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});
function tryRoute(packacge) {
    packacge.useRouter = jest.fn();
    packacge.useRouter.mockImplementation(() => ({ route: "/" }));
}
tryRoute(nextRouter);
