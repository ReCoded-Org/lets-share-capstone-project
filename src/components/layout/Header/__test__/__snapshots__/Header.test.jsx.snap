// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative "
>
  <div
    className=" absolute -z-10 h-96 w-full"
  />
  <nav
    className="bg-transperent "
  >
    <div
      className="max-w-auto  opacity-1 top-0 z-30 mb-2   px-2 transition duration-1000   ease-in-out sm:px-3 lg:px-4 false"
    >
      <div
        className=" flex items-center justify-end"
      >
        <div
          className="absolute flex items-center md:hidden"
        >
          <button
            aria-expanded={false}
            className=" hover:text-slate-900 focus:ring-white inline-flex items-center justify-center rounded-md p-2 hover:bg-btnBgHover focus:outline-none focus:ring-2 focus:ring-inset"
            id="headlessui-disclosure-button-:r0:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              className="block h-10 w-10"
              fill="none"
              stroke="#97dbbf"
              strokeWidth={2}
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          </button>
          <div
            className="hidden sm:ml-6 md:block "
          >
            <div
              className="flex space-x-4 "
            >
              <a
                className=" hover:bg-[gray-700] hover:text-white text-[white] rounded-md px-3 py-2 font-medium "
                href="/home"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Home
              </a>
              <a
                className=" hover:bg-[gray-700] hover:text-white text-[white] rounded-md px-3 py-2 font-medium "
                href="/aboutus"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                About us
              </a>
              <a
                className=" hover:bg-[gray-700] hover:text-white text-[white] rounded-md px-3 py-2 font-medium "
                href="/goods"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Goods
              </a>
              <a
                className=" hover:bg-[gray-700] hover:text-white text-[white] rounded-md px-3 py-2 font-medium "
                href="/blogs"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Blogs
              </a>
            </div>
          </div>
        </div>
        <img
          alt="Lets share logo"
          className="h-14 w-auto"
          src="/images/dummyLogo.svg"
        />
        <div
          className="flex flex-1 items-center justify-center sm:items-stretch sm:justify-end "
        >
          <div
            className="flex flex-shrink-0 items-center"
          >
            <div
              className="hidden sm:ml-6 md:block "
            >
              <div
                className="mr-5 flex space-x-4 "
              >
                <a
                  className="hover:text-white mx-1 rounded-md px-3 py-2 text-lg font-medium text-fontColor hover:bg-btnBgHover "
                  href="/home"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Home
                </a>
                <a
                  className="hover:text-white mx-1 rounded-md px-3 py-2 text-lg font-medium text-fontColor hover:bg-btnBgHover "
                  href="/aboutus"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  About us
                </a>
                <a
                  className="hover:text-white mx-1 rounded-md px-3 py-2 text-lg font-medium text-fontColor hover:bg-btnBgHover "
                  href="/goods"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Goods
                </a>
                <a
                  className="hover:text-white mx-1 rounded-md px-3 py-2 text-lg font-medium text-fontColor hover:bg-btnBgHover "
                  href="/blogs"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Blogs
                </a>
              </div>
            </div>
            <div
              className="hidden md:block "
            >
              <div
                className="relative ml-3"
              >
                <div>
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="mr-4 mb-2 flex rounded-full text-sm md:mr-1 md:mb-0 "
                    id="headlessui-menu-button-:r2:"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    type="button"
                  >
                    <span
                      className="sr-only"
                    >
                      Open user menu
                    </span>
                    <imge
                      alt=""
                      className="h-8 w-8 rounded-full"
                      src="/images/noUserPic.svg"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              className="hidden md:block"
            >
              <div
                className="relative ml-3"
              >
                <div>
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className=" z-1 mr-4 flex flex-row-reverse rounded-full text-sm md:mr-0 md:flex-row"
                    id="headlessui-menu-button-:r3:"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    type="button"
                  >
                    <span
                      className="sr-only"
                    >
                      Open user menu
                    </span>
                    <img
                      alt=""
                      className="h-8 w-8 rounded-full "
                      src="/images/undefinedFlag.svg"
                    />
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "inline-block",
                          "height": "initial",
                          "margin": 0,
                          "maxWidth": "100%",
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "relative",
                          "width": "initial",
                        }
                      }
                    >
                      <span
                        style={
                          Object {
                            "background": "none",
                            "border": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": "initial",
                            "margin": 0,
                            "maxWidth": "100%",
                            "opacity": 1,
                            "padding": 0,
                            "width": "initial",
                          }
                        }
                      >
                        <img
                          alt=""
                          aria-hidden={true}
                          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2730%27/%3e"
                          style={
                            Object {
                              "background": "none",
                              "border": 0,
                              "display": "block",
                              "height": "initial",
                              "margin": 0,
                              "maxWidth": "100%",
                              "opacity": 1,
                              "padding": 0,
                              "width": "initial",
                            }
                          }
                        />
                      </span>
                      <img
                        className="arrow rotate-90 md:rotate-0 "
                        data-nimg="intrinsic"
                        decoding="async"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          className="arrow rotate-90 md:rotate-0 "
                          data-nimg="intrinsic"
                          decoding="async"
                          loading="lazy"
                          src="/images/arrowIcon.svg"
                          srcSet="/images/arrowIcon.svg 1x, /images/arrowIcon.svg 2x"
                          style={
                            Object {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
